src/mergeQueue.ts

```diff
@@ -10,19 +10,22 @@ export const run = async () => {
   const pendingMerge = await fetchPendingMerge();
 
   // group PRs by milestone
-  const milestoneToPr = new Map<string, number[]>();
+  const milestoneToPr = new Map<
+    string,
+    { number: number; user: { login: string } }[]
+  >();
   for (const pr of pendingMerge.items) {
     const milestone = pr.milestone?.title;
     const prs = milestoneToPr.get(milestone) ?? [];
-    prs.push(pr.number);
+    prs.push(pr);
     milestoneToPr.set(milestone, prs);
   }
 
   // for each milestone, try to update the lowest PR number (only if it needs an update), if it fails, try the next one
   for (const [_, prs] of milestoneToPr) {
     for (const pr of prs) {
-      if (!await needsUpdate(pr)) break;
-      const response = await updatePr(pr);
+      if (!await needsUpdate(pr.number)) break;
```

Comment:
Shouldn't typescript prevent these class of error, e.g. calling a function with incorrect arguments? :)

Reply:
I changed PR from a number to an object in this change. To keep the author information

Reply:
ah, yes